apps/overige-objecten-api/src/controllers/objects/update.ts

```diff
@@ -0,0 +1,110 @@
+import type { RequestHandler } from 'express';
+import merge from 'lodash.merge';
+import snakeCase from 'lodash.snakecase';
+import { GET_VAC_ITEM_BY_UUID, UPDATE_VAC } from '../../queries';
+import type { DataVacItem, VACSData } from '../../strapi-product-type';
+import type { components } from '../../types/openapi';
+import { fetchData, getCurrentTypeParam, getTheServerURL } from '../../utils';
+
+type VACData = {
+  data: {
+    updateVac: {
+      data: DataVacItem;
+    };
+  };
+};
+
+export const updateVacController: RequestHandler = async (req, res, next) => {
+  const body = req.body as components['schemas']['ObjectData'];
+  const vac = body?.record?.data as components['schemas']['vac'];
+  const isAuthHasToken = req.headers?.authorization?.startsWith('Token');
+  const tokenAuth = isAuthHasToken ? req.headers?.authorization?.split(' ')[1] : req.headers?.authorization;
+  const locale = req.query?.locale || 'nl';
+  const graphqlURL = new URL('/graphql', process.env.STRAPI_PRIVATE_URL);
+  const serverURL = getTheServerURL(req);
+  const uuid = req.params?.uuid;
+  const { isVac } = getCurrentTypeParam(body.type);
+
+  // Check if UUID is provided
+  if (!uuid) {
+    return res.status(404).json({ message: 'UUID not provided' });
+  }
+
+  try {
+    const vacUrl = new URL(`/api/v2/objects/${body?.uuid}`, serverURL).href;
+    const vacSchemaURL = new URL('api/v2/objecttypes/vac', serverURL).href;
+
+    if (!isVac) return res.status(400).json({ message: 'Type is not allowed' });
+
+    const { data: vacData } = await fetchData<{ data: VACSData }>({
+      url: graphqlURL.href,
+      query: GET_VAC_ITEM_BY_UUID,
+      variables: { uuid },
+      headers: {
+        Authorization: `Bearer ${tokenAuth}`,
+      },
+    });
+
+    if (vacData?.vacs?.data?.length === 0) return res.status(404).json({ message: 'Object not found' });
```

Comment:
Wat als `vacData` leeg is? Zou het niet veiliger zijn om te checken `if (!(vacData?.vacs?.data?.length > 0))` ofzo